### PR TITLE
docs: Add GitHub links to Type Providers + mention Zod

### PR DESCRIPTION
**Pull request by melroy89**
- Add links behind each package name, so people can easily find and look-up the corresponding READMEs
- Add Zod to the TypeScript list as 3rd party
- Add Type Provider wrapper packages to the Type Provider page
- Make all links consistency, not using "GitHub" anymore

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

File: docs/Reference/TypeScript.md [hunk 1]
```diff
@@ -207,21 +207,23 @@ Here are some options on how to achieve this.
 
 Fastify offers two packages wrapping `json-schema-to-ts` and `typebox`:
 
-- `@fastify/type-provider-json-schema-to-ts`
-- `@fastify/type-provider-typebox`
+- `@fastify/type-provider-json-schema-to-ts` - [GitHub](https://github.com/fastify/fastify-type-provider-json-schema-to-ts)
```
Comment (Fdawgs) [inline docs/Reference/TypeScript.md:210]: Just add the links to the list items rather than having a separate "GitHub" bit?

**Review by Fdawgs — CHANGES_REQUESTED** [review]

File: docs/Reference/TypeScript.md (hunk 1 above)
Comment (melroy89) [reply docs/Reference/TypeScript.md:210]: Sure I can do that. But then I will also update the type provider doc page.

Comment (melroy89) [reply docs/Reference/TypeScript.md]: @Fdawgs I updated the docs with your feedback and like I mentioned, I also updated the Type provider page to keep all links consistent.

**Review by mcollina — APPROVED** [review]
lgtm

**Review by jsumners — APPROVED** [review]

File: docs/Reference/TypeScript.md [hunk 2]
```diff
@@ -202,21 +202,23 @@ Here are some options on how to achieve this.
 
 Fastify offers two packages wrapping `json-schema-to-ts` and `typebox`:
 
-- `@fastify/type-provider-json-schema-to-ts`
-- `@fastify/type-provider-typebox`
+- [`@fastify/type-provider-json-schema-to-ts`](https://github.com/fastify/fastify-type-provider-json-schema-to-ts)
+- [`@fastify/type-provider-typebox`](https://github.com/fastify/fastify-type-provider-typebox)
+
+And a `zod` wrapper by a third party called: [`fastify-type-provider-zod`](https://github.com/turkerdev/fastify-type-provider-zod)
```
Comment (Fdawgs) [inline docs/Reference/TypeScript.md:208]:
```suggestion
And a `zod` wrapper by a third party called [`fastify-type-provider-zod`](https://github.com/turkerdev/fastify-type-provider-zod)
```

File: docs/Reference/Type-Providers.md [hunk 3]
```diff
@@ -17,10 +17,15 @@ convention, and there are several community ones available as well.
...
-- `zod` - [github](https://github.com/colinhacks/zod)
+- [`json-schema-to-ts`](https://github.com/ThomasAribart/json-schema-to-ts)
+- [`typebox`](https://github.com/sinclairzx81/typebox)
+- [`zod`](https://github.com/colinhacks/zod)
+
+See also the Type Provider wrapper packages for each of the packages respectively:
+
+- [`@fastify/type-provider-json-schema-to-ts`](https://github.com/fastify/fastify-type-provider-json-schema-to-ts)
+- [`@fastify/type-provider-typebox`](https://github.com/fastify/fastify-type-provider-typebox)
+- [`fastify-type-provider-zod`](https://github.com/turkerdev/fastify-type-provider-zod)
```
Comment (Fdawgs) [inline docs/Reference/Type-Providers.md:28]: Think we need to make it clear this is a third party type provider

**Review by Fdawgs — CHANGES_REQUESTED** [review]

File: docs/Reference/Type-Providers.md (hunk 3 above)
Comment (melroy89) [reply docs/Reference/Type-Providers.md:28]: Wrong place I think..

**Comment by melroy89** [comment]
Owno I made the linter mad

File: docs/Reference/Type-Providers.md (hunk 3 above)
Comment (melroy89) [reply docs/Reference/Type-Providers.md:28]: ... The linter doesn't like this.. It's too long already. What did I do... 😢

**Review by mcollina — APPROVED** [review]
lgtm

**Review by Fdawgs — APPROVED** [review]

**Comment by melroy89** [comment]
Thanks for approving.

**Pull request merged**